parser/regexp.go

```diff
@@ -0,0 +1,100 @@
+package parser
+
+import (
+	"fmt"
+	"regexp"
+	goregexp "regexp"
+	"strings"
+
+	regexp2 "github.com/dlclark/regexp2"
+)
+
+type Regexp struct {
+	expression string
+	regexpGo *goregexp.Regexp
+	regexpLib *regexp2.Regexp
+}
+
+func NewRegexp(re string) *Regexp {
+	r := &Regexp{expression: re}
+	r.MustCompile()
+	return r
+}
+
+func (r *Regexp) compileGoRegex() error {
+		// Make some adjustments for a different regex engine than upstream matomo
```

Comment:
It looks like all of the adjustments are the same for both Golang regexp and regexp2. It'd be good to DRY this out. You could perform this adjustment once from the `Compile` function pass the adjusted expression into each of the implementation-specific compile functions.